member-service/src/main/java/com/hedvig/memberservice/services/ReviveMemberService.kt

```diff
@@ -0,0 +1,161 @@
+
+        log.info("Replaying {} events for member", events.size)
+
+        for (event in events) {
+
+            log.info("Execute '{}'", event.payloadType.simpleName)
+
+            when (event.payloadType.simpleName) {
+                "MemberCreatedEvent" -> memberEventListener.on(event.payload as MemberCreatedEvent)
+                "SSNUpdatedEvent" -> memberEventListener.on(event.payload as SSNUpdatedEvent)
+                "NorwegianSSNUpdatedEvent" -> memberEventListener.on(event.payload as NorwegianSSNUpdatedEvent)
+                "DanishSSNUpdatedEvent" -> memberEventListener.on(event.payload as DanishSSNUpdatedEvent)
+                "TrackingIdCreatedEvent" -> memberEventListener.on(event.payload as TrackingIdCreatedEvent)
+                "LivingAddressUpdatedEvent" -> memberEventListener.on(event.payload as LivingAddressUpdatedEvent)
+                "MemberStartedOnBoardingEvent" -> memberEventListener.on(event.payload as MemberStartedOnBoardingEvent)
+                "MemberInactivatedEvent" -> memberEventListener.on(event.payload as MemberInactivatedEvent)
+                "MemberCancellationEvent" -> memberEventListener.on(event.payload as MemberCancellationEvent)
+                "PhoneNumberUpdatedEvent" -> memberEventListener.on(event.payload as PhoneNumberUpdatedEvent)
+                "BirthDateUpdatedEvent" -> memberEventListener.on(event.payload as BirthDateUpdatedEvent)
+                "FraudulentStatusUpdatedEvent" -> memberEventListener.on(event.payload as FraudulentStatusUpdatedEvent)
+                "AcceptLanguageUpdatedEvent" -> memberEventListener.on(event.payload as AcceptLanguageUpdatedEvent)
+                "PickedLocaleUpdatedEvent" -> memberEventListener.on(event.payload as PickedLocaleUpdatedEvent)
+                "NewCashbackSelectedEvent" -> memberEventListener.on(event.payload as NewCashbackSelectedEvent)
+                "MemberIdentifiedEvent" -> identityEventListener.on((event.payload as MemberIdentifiedEvent))
+                "EmailUpdatedEvent" -> {
+                    memberEventListener.on(event.payload as EmailUpdatedEvent)
+                    customerIOEventListener?.on((event.payload as EmailUpdatedEvent))
+                }
+                "NameUpdatedEvent" -> {
+                    memberEventListener.on(event.payload as NameUpdatedEvent)
+                    customerIOEventListener?.on((event.payload as NameUpdatedEvent))
+                }
+                "MemberSignedEvent" -> {
+                    var signEvent = event.payload as MemberSignedEvent
+
+                    if (signEvent.ssn == null) {
+
+                        val ssn = events.reversed().stream()
+                            .filter { it.payloadType.simpleName == "SSNUpdatedEvent" }
+                            .map { it.payload as SSNUpdatedEvent }
+                            .map { it.ssn }
+                            .filter { it != null }
+                            .findFirst()
```

Comment:
I think there exists something like
```suggestion
                            .firstNotNull()
```

Reply:
Yeah, I have seen a `.filterNotNull()` somewhere else but I cannot find anything like it on the stream using code completion...